Activities

Victorma opened the pull request:
+ Changed sessions to activities
+ Classes are no longer needed
+ Toolbar access fixed
+ Routes updated
+ Improved visuals
+ And many other things...

Victorma committed:
Added fast-setup win

Victorma committed:
Classes changed to be independent

Victorma committed:
Game view to show activities

Victorma committed:
Activity controllers

Victorma committed:
Main controller, services, home, routes...

Victorma committed:
Working toolbar

Victorma committed:
Minor reference changes

Victorma committed:
Creation placeholders and errors feedback

gorco commented on app/public/js/controllers/activities.js:
```diff
@@ -0,0 +1,245 @@
+            $scope.$storage = $localStorage;
+            $scope.activity = {};
+            $scope.class = {};
+            $scope.game = {};
```
$scope.game and $scope.class are not used in this controller

Victorma replied:
Removed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/572bff82cfdb94cdb0df0d32cef2344ce44ca856

gorco commented on app/public/js/controllers/activity.js:
```diff
@@ -18,7 +18,7 @@
 
```
If we use Activities controller now, the old Activity controller has to dissapear

Victorma replied:
The activity-list controller manages the creation and deletion. The activity manages the modification and realtime start and stop.

gorco commented on app/views/layout.jade:
```diff
@@ -61,14 +61,14 @@ html
         script(src='js/controllers/data.js')
         script(src='js/controllers/analytics.js')
-        script(src='js/controllers/session.js')
+        script(src='js/controllers/activity.js')
```
we have to replace activity by activities (remove the activity reference)

Victorma replied:
Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/572bff82cfdb94cdb0df0d32cef2344ce44ca856 and https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/232f3d0b7236dd1f7e4f23a330d256a0370a11aa

gorco commented on app/views/toolbar.jade:
```diff
@@ -41,15 +41,24 @@ nav.navbar.navbar-inverse.navbar-top
+                                a(ng-if='!isDeveloper()' ng-href='gameactivity?game={{game._id}}' ng-click='setSelectedGame(game)') {{game.title}}
                 li.dropdown
-                    li(ng-if='isUser()&&isTeacher()&&hasSessions()')
+                    li(ng-if='isUser()&&hasClasses()')
```
we should check the role (we don't want that developers to see classes)

Victorma replied:
Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/5353294c7dba6c4720c637a730bf89ca7c7cbfb0

gorco commented on app/views/toolbar.jade:
```diff
@@ -41,15 +41,24 @@ nav.navbar.navbar-inverse.navbar-top
-                            li(ng-repeat='session in sessions')
-                                a(ng-href="#" ng-click='setSelectedSession(session, "data")') {{session.name}}
+                            li(ng-repeat='class in classes')
+                                a(ng-if='isDeveloper()' ng-href="#" ng-click='setSelectedClass(class, "data")') {{class.name}}
```
The developers can't see classes

Victorma replied:
Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/5353294c7dba6c4720c637a730bf89ca7c7cbfb0

gorco commented on app/views/data/realtime/realtime.jade:
```diff
@@ -16,7 +16,7 @@
  
 link(rel="stylesheet", href="css/realtime.css")
-.container-fluid.height100(ng-controller="SessionCtrl" role="tabpanel" style="position:relative")
+.container-fluid.height100(ng-controller="ActivityCtrl" role="tabpanel" style="position:relative")
```
ActivitiesCtrl

Victorma replied:
Fixed in: https://github.com/e-ucm/rage-analytics-frontend/pull/90/commits/0edb046efc1121fa233acb2cd3d9afd7dd93ff4f

gorco reviewed (changes requested):
Great work, some commets pending.
Awesome changes at home view

Victorma committed:
Changed router too return view chunks

Victorma committed:
Added angular-ui-router to bower

Victorma committed:
Added ng-router and updated layout

Victorma committed:
Removed loading from app.js

Victorma committed:
Moved class things from app.js to class.js

Victorma committed:
Moved game things from app.js to game.js

Victorma committed:
Moved Kibana code from app.js to kibana.js

Victorma committed:
Moved analysis things from app.js to analysis.js

Victorma committed:
Activity students management moved

Victorma committed:
Moved LTI to class.js

Victorma committed:
Renamos activities to activity-list and moved things from app.js

Victorma committed:
Added teacher management

Victorma committed:
Updated services to use methods

Victorma committed:
Moved allow anonymous to activity.js

Victorma committed:
Moved change activity name

Victorma committed:
Moved developer management to game.js

Victorma committed:
Moved alert and warning management from data.js to game.js

Victorma committed:
Removed loading stuff and selecting stuff in appjs
- On the load phase now only the selected stuff is loaded
- Selection is now made through events 'selectClass' 'selectGame' and 'selectActivity'

Victorma committed:
Created ToolbarCtrl Moved toolbar funcionallities to toolbar.js

Victorma committed:
Created class-list controller to manage the classes

Victorma committed:
Moved home.js to game-list.js and added deletion from app.js

Victorma committed:
Fixed login reload

Victorma committed:
Fixed login part 2

Victorma committed:
Game js initialization

Victorma committed:
Class js initialization

Victorma committed:
Activity js intialization

Victorma committed:
Re-added realtime controls to activity js

Victorma committed:
Updated controllers from layout and updated references in app.js

Victorma committed:
Removed duplicated stuff present in activity-list.js

Victorma committed:
Moved all the views to view folder

This allows router to differ between the home angular render and the
view chunks

Victorma committed:
Removed blocks from view

Views are now independent chunks

Victorma committed:
Updated home to use new list controllers

Victorma committed:
Updated class and game view to use activitylist

Victorma committed:
Added analysis controller in storm analysis view

Victorma committed:
Added kibana controller in analysis view

Victorma committed:
Changed global to local references in data views

Victorma committed:
Fixes for students view

Victorma committed:
Added teacher's activity management

Victorma committed:
Added teachers class management

Victorma committed:
Added controller's initialization
- And fixed line endings

Victorma committed:
Added loading icon

Victorma committed:
Changed activitiesctrl to activitylistctrl

Victorma committed:
Changed classesctrl to classlistctrl

Victorma committed:
Changed gamesctrl to gamelistctrl

Victorma committed:
Fixed health test

Victorma committed:
Limitations in toolbar

gorco commented:
I'm testing the front, but when I create a new game (also if I open a game) the system throws these errors...
Backend:
![backerror](https://user-images.githubusercontent.com/5657407/27530238-1aa2e6e8-5a58-11e7-8faa-48385d96eb05.png)
Frontend
![fronterror](https://user-images.githubusercontent.com/5657407/27530241-1ca90486-5a58-11e7-99b3-a8093b430338.png)

gorco commented:
The developer visualizations don't load
![frontdev](https://user-images.githubusercontent.com/5657407/27531393-b87a4e2a-5a5c-11e7-80ef-075781db2cd9.png)

gorco commented:
As teacher, Into a Game, if I try create a activity, I can't because the frontend says that I don't have selected a game.
![fronteaerror](https://user-images.githubusercontent.com/5657407/27531564-53fa14f2-5a5d-11e7-82d0-04663a5488db.png)

Victorma committed:
Fixed showing activity and class in data.jade

This removes the loading of activity and class controllers in developer
view

Victorma committed:
Case sensitive

Victorma committed:
Fixed loading button
